mslice/models/workspacemanager/mantid_workspace_provider.py

```diff
@@ -166,9 +168,9 @@ def get_limits(self, workspace, axis):
 
     def propagate_properties(self, old_workspace, new_workspace):
         """Propagates MSlice only properties of workspaces, e.g. limits"""
-        if old_workspace in self._EfDefined.keys():
+        if old_workspace in self._EfDefined:
             self._EfDefined[new_workspace] = self._EfDefined[old_workspace]
-        if old_workspace in self._limits.keys():
+        if old_workspace in self._limits:
```

Comment:
@MatthewBowles Do you know why the `.keys` error here that was fixed in the last commit wasn't picked up by a test?

Reply:
@martyngigg Looks like the only classes that use it are mocked out in the tests

Reply:
I see. Would it be hard to write some tests for it?

Reply:
Seems doable. Should I open a separate issue?

Reply:
I think that would be a good plan. You can never have too much test coverage!